hal/src/electron/cellular_hal.cpp

```diff
@@ -7,6 +8,36 @@
 
 static CellularCredentials cellularCredentials;
 
+static HAL_NET_Callbacks netCallbacks = { 0 };
+
```

Comment:
If we need new callbacks yes...but if we need to extend the existing callbacks with additional parameters, we can just add them to the callback. Older system modules won't see the extra parameters, so that has to be kept in mind, but since they are informational/status that's not a concern - they only mean something if the receiver can understand them.